### PR TITLE
Fix some issues with going global, add /vol to change radio volume.

### PR DESCRIPTION
**Pull request by AvarianKnight**
Mumble uses the 'grid' system which I haven't been able to find out why it was necessary as it causes a lot more issues than it resolves.

This commit makes the grid the entire map. 
Changes certain keybinds to use RegisterKeyMapping instead of looping.
Removes the watermark (you can feel free to add it back if you want to, I just personally dislike watermarks)
Adds a /vol command to reduce the volume of the radio

### CODE REVIEW

**Comment by kevinmlsilva** [comment]
You're using ESX specific method, and I guess it should be global / Native, and why did you removed the range for the voice / the possibility to change with a keybinding? not radio? o

**Comment by FrazzIe** [comment]
- I like the parts where you cleaned up the duplicate voiceData checks etc
- I don't like the implementation of RegisterKeyMapping, i believe it could be made a lot more configurable for the normies, and I previously had mapping in before but at the time they didn't always work.
- The watermark, doesn't bother me, but instead of completely removing it I believe it could be made a configurable option
- You included your specifc server config, which is annoying tbh
- You also changed the radio list default format (I'd rather leave it as is, you are free to change whatever you want)
- I don't agree with removing the grid, if it works for you, it works for you.
- The changes made to how the grid channel targets are set I do agree with, might even be the reason for one of the bugs going around
- Was there a specific reason you disabled the 2dAudioInVehicles check? is that a cause of one of the github issues?
- You left some esx bullshit in the client, and some weird trigger event for the radioActive
- You also added comments in the code to why you changed something, that's the whole point of a github commit, to say what you changed and why?

TL;DR
I like some of the changes but i'm not gonna pr this.

**Comment by AvarianKnight** [comment]
Sorry, I thought I had removed the ESX portion that was in it, it should be completely removed now.

I've used the RegisterKeyMapping's for a while and haven't had any issues, so far it seems to have cleaned up the loops.

I removed the server-specific config, sorry about that.

The Trigger events were also leftover code from stuff that's specifically for my server, that will be removed in one of the following commits.

I'm sorry if this came off as 'fuck your grid system', but the grid system for me has caused a lot of issues (like the not being able to talk to either on cars, players not being able to hear each other, players going global for no reason).

^ This is the reason I set 2dAudioInVehicles to false, as using a much bigger grid makes this not a problem anymore.

**Comment by AvarianKnight** [comment]
@kevinmlsilva They're not removed, you're able to set them in your keybinds (Fivem->Settings)

**Comment by AvarianKnight** [comment]
This should fix #95 #92 #82 #79 and add's feature #83

**Comment by ghost** [comment]
Afew things ive noticed is you removed the controls from the config

so speaking on the radio with caps no longer works
and trying to set the speaking distance from shouting to whispering ect usually done with z also no longer works
and setting the watermark to false in the config does not remove it.

**Comment by AvarianKnight** [comment]
@ONYX-X The default keys changed with this (you'll have to change them in your FiveM settings) I'll add a way to configure this so you can set your default keys.

**Comment by ghost** [comment]
My bad :D to early in the morning for me haha.

great work man honestly we were having the global issue for awhile.

**Comment by AvarianKnight** [comment]
@ONYX-X I understand that one, I'm about to go pass out 🤣 latest commit should fix the issues you described.

**Comment by nnsdev** [comment]
@AvarianKnight We tried to use your fork (although your original commit, not tried after your changes yet), we have realized an issue with phone calls. It seems like when you call somebody that is currently in your "grid"(?), others that are in the same grid can hear you, but you cannot hear them anymore, including the other person you called with.

The issue is solved by running out of the "grid", and back in. We tried to look for it briefly yesterday, but could not isolate the issue just yet. Maybe you know, what could be the cause of this. Script was basically run completely original, without any changes of us.

Wanted to throw this in before this gets merged, I am not quite sure if this is already happening on the current mumble-voip latest, I did not get to test that yet.

**Comment by AvarianKnight** [comment]
@JulienGelmar If you could find any reproduction steps that would be great, I haven't had it happen on the server I develop for so unless I get repro I can't do much.

**Comment by AvarianKnight** [comment]
I keep hitting the close with comment button 🤣

**Comment by nnsdev** [comment]
@AvarianKnight 
The way we tested it is the following:
- Have 4 people close to each other
- 2 people call each other (here done over gcphone), speak, it'll work all fine, others around you can hear you speak and you can hear them
- Hang up, now you cannot hear anybody anymore (not even the person you called), but everybody can hear you
- Walk out of range, come back, it will reset and everything it back to normal
- The other two people standing around can keep talking, nothing will change for them, but the 2 caller parties just wont hear them
- This is **not** happening with radio, there it all works.

**Comment by AvarianKnight** [comment]
@JulienGelmar Can you send me how you're doing all of it in GCPhone? I can't reproduce it with how my server does it (we use 8bit_phone)

**Comment by nnsdev** [comment]
Okay disregard my issues, gcphone was invoking a native by its hex instead of the native itself, so I didn't notice it...That is my bad, sorry for false reporting.

**Comment by AvarianKnight** [comment]
@JulienGelmar You're fine, sometimes it happens 🤷‍♂️

**Comment by Patrickstek121** [comment]
I have still issue's when i speak to someone. I can hear everyone in the server no matter whats the range

**Comment by AvarianKnight** [comment]
@patr

> I have still issue's when i speak to someone. I can hear everyone in the server no matter whats the range

Do you have OneSync?

**Comment by Patrickstek121** [comment]
Yeah i have onesync

**Comment by AvarianKnight** [comment]
Do you have these set


```
set voice_use3dAudio true
set voice_useSendingRangeOnly true
```

**Comment by Patrickstek121** [comment]
image

or in the server.cfg?
because this is the mumble voip config on screen

**Comment by Patrickstek121** [comment]
added to server.cgf but still the same issue that i can hear someone

**Comment by AvarianKnight** [comment]
This doesn't 'fix' the issue, was just a bandage (which broke when over 120+ players).
